v2/protocol/http/protocol.go

```diff
@@ -218,6 +223,7 @@ type msgErr struct {
 func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	m := NewMessageFromHttpRequest(req)
 	if m == nil || m.ReadEncoding() == binding.EncodingUnknown {
+		http.Error(rw, fmt.Sprintf("Cannot read CloudEvent: %s", binding.ErrUnknownEncoding), http.StatusBadRequest)
```

Comment:
Can we address this problem in the https://github.com/cloudevents/sdk-go/pull/480?

Reply:
Ah ok i see now, the problem is different from #480.

Anyway, I think this is a bad change because `EncodingUnknown` happens in several situations, even on a simple get request without any parameter/cloudevent: eg ff the receiver just wants to return events, without expecting to receive any, the encoding of the received event will be unknown. What i'm trying to say is: at the protocol level, i think we should not just fail the request, this should be up to the protocol user. If you experienced this problem using the client, then we need to address the problem in a different way on the client code.

Reply:
What I want to avoid here is external clients (not CE) sending random requests to the endpoint and crashing it. The HTTP response is to me a plus, just to play nice with the HTTP protocol. If that's not fine with the CE contract I'm happy to remove this.

Still, I'm a bit puzzled when I see a request fail with 200 and no payload in general.

Reply:
Removed in 49fae7d

Reply:
not crashing sounds like a good thing 😄